From 70c2fb516136b7831b93fdcd90092ba59293a5c1 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Thu, 19 Jan 2023 13:53:13 +0100
Subject: [PATCH] u-boot-qoriq: LS208xARDB Cortina check was broken

The personality bits and socketed RDBs permit using all family
members in an RDB. For some reason the Cortina FW address check was
tied to specific SoC types only. Same things for the board version.

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 board/freescale/ls2080ardb/ls2080ardb.c | 47 ++++++++++++-------------
 1 file changed, 22 insertions(+), 25 deletions(-)

diff --git a/board/freescale/ls2080ardb/ls2080ardb.c b/board/freescale/ls2080ardb/ls2080ardb.c
index f5ebb934eb..a888142a72 100644
--- a/board/freescale/ls2080ardb/ls2080ardb.c
+++ b/board/freescale/ls2080ardb/ls2080ardb.c
@@ -240,34 +240,29 @@ int config_board_mux(int ctrl_type)
 
 ulong *cs4340_get_fw_addr(void)
 {
+	ulong cortina_fw_addr = CONFIG_CORTINA_FW_ADDR;
 #ifdef CONFIG_TFABOOT
 	struct ccsr_gur __iomem *gur = (void *)(CONFIG_SYS_FSL_GUTS_ADDR);
 	u32 svr = gur_in32(&gur->svr);
-#endif
-	ulong cortina_fw_addr = CONFIG_CORTINA_FW_ADDR;
 
-#ifdef CONFIG_TFABOOT
-	/* LS2088A TFA boot */
-	if (SVR_SOC_VER(svr) == SVR_LS2088A) {
-		enum boot_src src = get_boot_src();
-		u8 sw;
-
-		switch (src) {
-		case BOOT_SOURCE_IFC_NOR:
-			sw = QIXIS_READ(brdcfg[0]);
-			sw = (sw & 0x0f);
-			if (sw == 0)
-				cortina_fw_addr = CORTINA_FW_ADDR_IFCNOR;
-			else if (sw == 4)
-				cortina_fw_addr = CORTINA_FW_ADDR_IFCNOR_ALTBANK;
-			break;
-		case BOOT_SOURCE_QSPI_NOR:
-			/* Only one bank in QSPI */
-			cortina_fw_addr = CORTINA_FW_ADDR_QSPI;
-			break;
-		default:
-			printf("WARNING: Boot source not found\n");
-		}
+	enum boot_src src = get_boot_src();
+	u8 sw;
+
+	switch (src) {
+	case BOOT_SOURCE_IFC_NOR:
+		sw = QIXIS_READ(brdcfg[0]);
+		sw = (sw & 0x0f);
+		if (sw == 0)
+			cortina_fw_addr = CORTINA_FW_ADDR_IFCNOR;
+		else if (sw == 4)
+			cortina_fw_addr = CORTINA_FW_ADDR_IFCNOR_ALTBANK;
+		break;
+	case BOOT_SOURCE_QSPI_NOR:
+		/* Only one bank in QSPI */
+		cortina_fw_addr = CORTINA_FW_ADDR_QSPI;
+		break;
+	default:
+		printf("WARNING: Boot source not found\n");
 	}
 #endif
 	return (ulong *)cortina_fw_addr;
@@ -351,7 +346,9 @@ int misc_init_r(void)
 	 * ls2080ardb. Modify board env for other supported SoCs
 	 */
 	if ((SVR_SOC_VER(svr) == SVR_LS2088A) ||
-	    (SVR_SOC_VER(svr) == SVR_LS2048A))
+	    (SVR_SOC_VER(svr) == SVR_LS2084A) ||
+	    (SVR_SOC_VER(svr) == SVR_LS2048A) ||
+	    (SVR_SOC_VER(svr) == SVR_LS2044A))
 		env_set("board", "ls2088ardb");
 	else if ((SVR_SOC_VER(svr) == SVR_LS2081A) ||
 	    (SVR_SOC_VER(svr) == SVR_LS2041A))
-- 
2.34.1

